### PR TITLE
update submit evaluation api

### PR DESCRIPTION
**Pull request by emmaacha**
Removed save evaluation api and added is_submitting flag for submit evaluation

Ticket ID: [Update API for SUBMIT for Performance Evaluation Form#116](https://github.com/nerubia/kh360-nodejs/issues/116)

FE PR: [update submit evaluation api #142](https://github.com/nerubia/kh360-react/pull/142)

### CODE REVIEW

File: src/controllers/user/user-controller.ts [hunk 1]
```diff
@@ -213,13 +213,14 @@ export const getEvaluationAdministrations = async (req: Request, res: Response)
  * @param req.body.evaluation_rating_ids - Evaluation rating ids.
  * @param req.body.answer_option_ids - Answer option ids.
  * @param req.body.comment - Evaluation comment.
+ * @param req.body.is_submitting - Evaluation comment.
```
Comment (jatnerubia) [inline src/controllers/user/user-controller.ts:216]: 🤔

**Review by jatnerubia — CHANGES_REQUESTED** [review]
^^

File: src/controllers/user/user-controller.ts (hunk 1 above)
Comment (emmaacha) [reply src/controllers/user/user-controller.ts:216]: Oops my bad 😅

File: src/repositories/evaluation-repository.ts [hunk 2]
```diff
@@ -35,3 +69,25 @@ export const updateStatusByAdministrationId = async (
...
+export const countAllByFilters = async (where: Prisma.evaluationsWhereInput) => {
+  const count = await prisma.evaluations.count({
+    where,
+  })
+
+  return count
+}
+
+export const aggregateSumByFilters = async (
+  _sum: Prisma.EvaluationsSumAggregateInputType,
```
Comment (jatnerubia) [inline src/repositories/evaluation-repository.ts:82]: we'r not using `_sum` here? 🤔

File: src/repositories/evaluation-result-detail-repository.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { type Prisma } from "@prisma/client"
+import prisma from "../utils/prisma"
+import { type EvaluationResultDetail } from "../types/evaluation-result-detail-type"
+
+export const getById = async (id: number) => {
+  return await prisma.evaluations.findUnique({
```
Comment (jatnerubia) [inline src/repositories/evaluation-result-detail-repository.ts:6]: 🤔

File: src/services/project-role-service.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import * as ProjectRole from "../repositories/project-role-repository"
```
Comment (jatnerubia) [inline src/services/project-role-service.ts:1]: use `ProjectRoleRepository`

File: src/types/evaluation-type.ts [hunk 5]
```diff
@@ -1,3 +1,19 @@
+export interface Evaluation {
```
Comment (jatnerubia) [inline src/types/evaluation-type.ts:1]: follow the order in schema

**Review by jatnerubia — CHANGES_REQUESTED** [review]
just a few changes ^^

File: src/repositories/evaluation-repository.ts (hunk 2 above)
Comment (emmaacha) [reply src/repositories/evaluation-repository.ts:82]: My bad, we should be passing and using it kuys, forgot to remove the _sum properties inside that function. Will update this thanks!

Comment (jatnerubia) [reply src/repositories/evaluation-repository.ts:82]: 🤔

File: src/services/evaluation-service.ts [hunk 6]
```diff
@@ -1,10 +1,26 @@
...
+
+export const getById = async (id: number) => {
+  return await EvaluationRepository.getById(id)
+}
 
 export const getAllByFilters = async (where: Prisma.evaluationsWhereInput) => {
   return await EvaluationRepository.getAllByFilters(where)
 }
 
+export const getAllDistictByFilters = async (
```
Comment (jatnerubia) [inline src/services/evaluation-service.ts:13]: typo 🤔

**Review by jatnerubia — CHANGES_REQUESTED** [review]
^^

File: src/repositories/evaluation-repository.ts (hunk 2 above)
Comment (emmaacha) [reply src/repositories/evaluation-repository.ts:82]: Oh, I think it was `aggregateSumById` I changed earlier, I thought it was this one. Will update this as well. My bad!

File: src/repositories/evaluation-result-detail-repository.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+export const updateById = async (id: number, data: EvaluationResultDetail) => {
+  await prisma.evaluation_result_details.update({
+    where: {
+      id,
+    },
+    data,
+  })
+}
+
+export const aggregateSum = async (_sum: Prisma.Evaluation_result_detailsSumAggregateInputType) => {
```
Comment (jatnerubia) [inline src/repositories/evaluation-result-detail-repository.ts:28]: I think we need to pass `evaluation_result_id` here

**Review by jatnerubia — CHANGES_REQUESTED** [review]
^^

**Review by jatnerubia — APPROVED** [review]
Nice

**Review by Dev-Nino — APPROVED** [review]

**Pull request merged**